core/code/ornaments.js

```diff
@@ -71,5 +139,109 @@ window.ornaments = {
       });
       delete this._portals[guid];
     }
+  },
+  initOrnaments: function () {
+    this.knownOrnaments = {};
+    this.save();
+  },
+
+  load: function () {
+    var dataStr;
+    try {
+      dataStr = localStorage['excludedOrnaments'];
+      if (dataStr === undefined) { return; }
+      this.excludedOrnaments = JSON.parse(dataStr);
+    } catch (e) {
+      console.warn('ornaments: failed to load excludedOrnaments from localStorage: '+e);
+    }
+    try {
+      dataStr = localStorage['knownOrnaments'];
+      if (dataStr === undefined) {
+        this.initOrnaments ();
+        return;
+      }
+      this.knownOrnaments = JSON.parse(dataStr);
+    } catch (e) {
+      console.warn('ornaments: failed to load data from localStorage: '+e);
+    }
+
+  },
+
+  save: function () {
+    localStorage['excludedOrnaments'] = JSON.stringify(this.excludedOrnaments);
+    localStorage['knownOrnaments'] = JSON.stringify(this.knownOrnaments);
+  },
+
+  // reload: addPortal also calls removePortal
+  reload: function () {
+    for (var guid in window.ornaments._portals) {
+      window.ornaments.addPortal(window.portals[guid]);
+    }
+  },
+
+  processInput: function () {
+    window.ornaments.excludedOrnaments = $('#ornaments_E').val().split(/[\s,]+/);
+    window.ornaments.excludedOrnaments = window.ornaments.excludedOrnaments.filter(function (ornamentCode) { return ornamentCode !== ''; });
+    // process the input from the checkboxes
+    for (var ornamentCode in window.ornaments.knownOrnaments) {
+      var input = document.getElementById('chk_orn_'+ornamentCode);
+      window.ornaments.knownOrnaments[ornamentCode] = input ? input.checked : false; // <- default value if the input is not found for unexpected reason
+    }
+  },
+
+  ornamentsList: function() {
+    var text ='';
+    var sortedIDs = Object.keys(window.ornaments.knownOrnaments).sort();
+
+    sortedIDs.forEach ( function (ornamentCode) {
+      var hidden = window.ornaments.excludedOrnaments.some( function(code) {
```

Comment:
so "excludedOrnaments" is always initilized? (see comment for line 107)

Reply:
Yes, I think so. It gets initialized in .setup.